src/main/java/net/minecraftforge/event/entity/living/VillagerEvent.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.entity.living;
+
+import net.minecraft.world.entity.npc.Villager;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.ForgeEventFactory;
+import net.minecraftforge.event.entity.EntityEvent;
+import net.minecraftforge.eventbus.api.Cancelable;
+
+/**
+ * VillagerEvent is fired for events related to villagers.
+ * If a method utilizes this event as its parameter, the method will
+ * receive every child event of this class.
+ *
+ * All children of this event are fired on the {@link MinecraftForge#EVENT_BUS}.
+ **/
+public class VillagerEvent extends EntityEvent {
```

Comment:
My opinion is that we should try to move away from this kind of event superclasses. While yes, they're consistent with how all other events currently are, in reality they don't make sense. It means people can simply to this event and cause weird behaviour any time a new event subclassing it is added.
Event inheritance is a flawed concept and 90% of the times someone listens to a super event it's a bug - not a feature.

Reply:
Would you recommend I create a `villager` package then?

Reply:
I think the Neo team should come to an agreement regarding the plan for events before such a change is acted.

Reply:
So what should I do in this PR?

Reply:
Per recent discussion in the Events thread in the Discord server (`neo-discussions > Events and EventBus`), we will be moving towards keeping superclass events (`VillagerEvent` in this situation), but marking them `abstract`.

In a future version of https://github.com/neoforged/Bus, event listeners will only receive the exact event type they receive (instead of receiving both the exact event class and all its subclasses), and attempts to listen to an `abstract` event will throw an exception.

In addition, per [a vote in the Discord server](https://canary.discord.com/channels/313125603924639766/852298000042164244/1139746854632955986), we are disfavoring the pattern of having a superclass event holds all its subevents as its inner classes (what is termed as the "event-innerclass pattern" in the linked message). As an example, the subevents in [`PlayerEvent`](https://github.com/neoforged/NeoForge/blob/1.20.x/src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java) will, in the future, be moved to the top-level in their own source files.

In line with the above:
- Mark `VillagerEvent` as `abstract`.
- Move the `ChangeProfessionEvent` into the top-level as its own source file.
- Move these two classes into a subpackage such as `net/minecraftforge/event/entity/villager`.

Alternatively, if there is little reason to believe that future events involving `Villager`s will be made, the `VillagerEvent` superevent can be removed instead (and thus not needing to move to its own dedicated subpackage).

Reply:
@sciwhiz12 we keep the inner classes per @cpw's veto.

Reply:
Unfortunately, I do not believe we recognize the ability of any team member to veto a discussion unilaterally, unless the steering council makes a resolution to overturn the decision of the maintainers.